core/src/main/java/org/bouncycastle/crypto/generators/OpenBSDBCrypt.java

```diff
@@ -12,7 +12,10 @@
  * Password hashing scheme BCrypt,
  * designed by Niels Provos and David Mazières, using the
  * String format and the Base64 encoding
- * of the reference implementation on OpenBSD
+ * of the reference implementation on OpenBSD.
+ * <p>
+ * Passwords are encoded using UTF-8. Encoded passwords longer than
```

Comment:
Regarding UTF-8: Do you think this wording here is misleading and it should say that this only applies when the password is provided as `char[]` (but not when provided as `byte[]`)? Or is it fine like this?

Reply:
I agree it's going to confuse some people. I've clarified the comment to say UTF-8 only applies for char[].